modules/gateway/gateway.go

```diff
@@ -105,27 +133,34 @@ func (g *Gateway) Info() (info modules.GatewayInfo) {
 }
 
 // New returns an initialized Gateway.
-func New(tcps *network.TCPServer, s *consensus.State) (g *Gateway, err error) {
-	if tcps == nil {
-		err = errors.New("gateway cannot use nil tcp server")
-		return
-	}
+func New(addr string, s *consensus.State) (g *Gateway, err error) {
 	if s == nil {
 		err = errors.New("gateway cannot use nil state")
 		return
 	}
 
-	g = &Gateway{
-		tcps:  tcps,
-		state: s,
-		peers: make(map[network.Address]struct{}),
-	}
-	block, exists := g.state.BlockAtHeight(0)
-	if !exists {
-		err = errors.New("gateway state is missing the genesis block")
+	// create listener
+	l, err := net.Listen("tcp", addr)
```

Comment:
Can we rename `l` to `listener` then? I'm just confused, because we call `net.Listen`, and then we call `go g.listen()` later on, and the whole process is a bit opaque.